rust/xaynet-analytics/src/data_combination/data_combiner.rs

```diff
@@ -247,15 +249,18 @@ mod tests {
                 PeriodUnit,
             },
         },
-        data_provision::{
-            analytics_event::{AnalyticsEvent, AnalyticsEventType},
-            data_provider::MockAnalyticsDataProvider,
+        database::{
+            analytics_event::{
+                data_model::{AnalyticsEvent, AnalyticsEventType},
+                repo::MockAnalyticsEventRepo,
+            },
+            common::MockRepo,
         },
     };
 
     #[test]
     fn test_init_screen_active_time_vars() {
-        let data_combiner = DataCombiner::new(MockAnalyticsDataProvider {});
+        let data_combiner = DataCombiner::new(MockAnalyticsEventRepo {}, MockRepo {});
```

Comment:
fyi (don't implemement it, just an idea for later)
I have a similar issue with our `Storage` trait
I also wanted to use a `MockStore` in our state machine tests and I first created a `DummyStore` manually  but then I remembered that we can also use `mockall`. It  basically automates the generation of a `MockStore` + it provides some handy methods like specifying the return value. I pushed the [branch](https://github.com/xaynetwork/xaynet/commit/ce7c817416bc712133c36f0f6e1684196c5e1fe5) if you want to take a look.

Reply:
thanks! yes i figured there would be some library for it. with Dart is called `mockito` i think